x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1243,6 +1248,25 @@ private Expression resolveConvertFunction(AbstractConvertFunction convert, List<
             return convert;
         }
 
+        private static Expression resolveFoldablesConvertFunction(FoldablesConvertFunction fcf) {
+            if (fcf.field() instanceof FieldAttribute fa && fa.field() instanceof InvalidMappedField imf) {
+                String unresolvedMessage = "argument of ["
+                    + fcf.sourceText()
+                    + "] must be a constant, received ["
+                    + Expressions.name(fa)
+                    + "]";
+                String types = imf.getTypesToIndices().keySet().stream().collect(Collectors.joining(","));
+                Expression ua = new UnsupportedAttribute(
```

Comment:
I believe we should use `UnresolvedAttribute` here.

`UnsupportedAttribute` enjoys privileged treatment by the verifier: it's used for
- fields with unsupported data type and
- multi-typed fields (that aren't resolved to a union type)
and are being passed through the compute engine as `null`s.

Currently, I think the way to mark something as invalid is using an unresolved attribute with a custom message.

(Maybe we should have an `InvalidAttribute` as well, because `UnresolvedAttribute` can either be "this is not resolved _yet_ but maybe later!", while we actually need something that says "we can't have a properly resolved attribute here because the query's invalid!")